### PR TITLE
Integrate w2v-bert2-LoRA-adapter-MFA model

### PR DESCRIPTION
**Pull request by shangguanqituan**
### Overview

This pull request (PR) integrates the advanced model proposed in the paper **"ENHANCING SPEAKER VERIFICATION WITH W2V-BERT 2.0 AND KNOWLEDGE DISTILLATION GUIDED STRUCTURED PRUNING"** into the `wespeaker` framework.

We have successfully implemented the full three-stage training pipeline for the `w2v-bert2-lora-adapter-mfa` model and ensured its compatibility with the existing `wespeaker` ecosystem. This not only introduces a powerful new model to the framework but also opens up new possibilities for future research and applications.

### Key Features and Changes

1.  **Model Integration**:
    * **Frontend**: Added a `w2vbert2` frontend in `wespeaker/frontend`, which incorporates **LoRA** (Low-Rank Adaptation) for efficient fine-tuning.
    * **Model**: Implemented an **Adapter-MFA** (Multi-Factor Attention) module in `wespeaker/models`. This module serves as the speaker model and is specifically designed to process the multi-layer hidden states from the `w2vbert2` frontend.

2.  **New Training Pipeline**:
    * Added complete **three-stage training configuration files (YAML)** and a corresponding execution script, `run_w2v.sh`, to facilitate the reproduction of the original paper's training process.

3.  **Framework Adaptations**:
    * **Dataflow Handling**:
        * To meet the input requirements of the MFA module, the `w2vbert2` frontend now returns all Transformer layer hidden states (`all_hidden_states`) as a tuple.
        * To handle this new return type (which differs from other frontends that return a `last_hidden_state` tensor), we have added conditional logic in the `executor` and `extract` modules to ensure smooth pipeline execution.
    * **DistributedDataParallel (DDP) Configuration**: Due to the increased complexity of gradient computation introduced by LoRA and MFA, we found it necessary to set `find_unused_parameters=True` for `DistributedDataParallel` in `train.py` to resolve gradient synchronization issues.
    * **Learning Rate Schedulers**: Added two new schedulers, `WarmupLR_withStepDecay` and `WarmupCosineScheduler`, to `scheduler.py` to meet the specific requirements for reproducing the paper's results.
    * **ASP Compatibility**: The Automatic Speaker Verification Pipeline (ASP) has been slightly modified to be compatible with the new model's outputs without affecting the functionality of existing models in the framework.

4.  **Pre-trained Model Compatibility**:
    * To allow users to easily load the official pre-trained checkpoints (ckpt) from the paper, we have modified `checkpoint.py`.
    * This change addresses a classifier dimension mismatch (`5994*3` vs. `5994`) that occurs because the original paper's third training stage does not use `speed_perturb` data augmentation. The code can now intelligently slice the weights to the correct dimension, enabling successful loading of the official model.

We believe this integration will greatly enrich the `wespeaker` model zoo and provide the community with a powerful new tool. We look forward to your feedback and review!

### CODE REVIEW

**Review by wsstriving — COMMENTED** [review]
Can you also update the results and the pretrained model pages?

File: .pre-commit-config.yaml
Comment (cdliang11) [inline .pre-commit-config.yaml]: Why modify this file?

File: examples/sre/v3/local/make_system_sad.py
Comment (cdliang11) [inline examples/sre/v3/local/make_system_sad.py]: To ensure the simplicity of PR, irrelevant parts do not need to be modified.

Comment (shangguanqituan) [reply examples/sre/v3/local/make_system_sad.py]: I didn’t intentionally modify those two files — it seems the formatting tool automatically updated them during commit. I will submit a clean PR later.

File: examples/voxceleb/v2/run_w2v.sh [hunk 1]
```diff
@@ -0,0 +1,273 @@
+#!/bin/bash
+
+# Copyright 2025 Your Name/Org (your_email@example.com)
```
Comment (cdliang11) [inline examples/voxceleb/v2/run_w2v.sh]: Change to your name and email address

File: wespeaker/frontend/w2vbert.py [hunk 2]
```diff
@@ -0,0 +1,388 @@
+# Copyright (c) 2025 Your Name/Org (your_email@example.com)
```
Comment (cdliang11) [inline wespeaker/frontend/w2vbert.py]: Change to your name and email address

File: wespeaker/models/w2vbert_adapter_mfa.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
+# Copyright (c) 2025 Your Name/Org
```
Comment (cdliang11) [inline wespeaker/models/w2vbert_adapter_mfa.py]: Change to your name and email address

**Comment by cdliang11** [comment]
> @cdliang11 maybe we also want to support this model in the CLI mode? @shangguanqituan We also need to decide which checkpoint to upload in this mode, the one you trained or the one provided by the original paper

Of course, supporting this model in CLI mode is feasible. @shangguanqituan, please select the model checkpoint to adopt.
